src/modules/fancyzones/lib/FancyZones.cpp

```diff
@@ -867,10 +867,8 @@ void FancyZones::UpdateDragState(HWND window, require_write_lock) noexcept
         m_dragEnabled = !(shift | mouse);
     }
 
-    const bool windowElevated = IsProcessOfWindowElevated(window);
-    static const bool meElevated = is_process_elevated();
     static bool warning_shown = false;
-    if (!meElevated && windowElevated)
+    if (!is_process_elevated() && IsProcessOfWindowElevated(window))

```

Comment:
I meant to check `warning_shown` and `is_cant_drag_elevated_warning_disabled` before checking `is_process_elevated()` and `IsProcessOfWindowElevated(window)`

Reply:
we can't do that, because we still need to do `m_dragEnabled = false;`

Reply:
I see, I guess we should consider adding a cache to avoid executing IsProcessOfWindowElevated() every time for the same windows, it's not high priority now, but please open an issue to track it.

`is_process_elevated()` is not cached, correct? It should since it can't change for the current process.

Reply:
opened an issue - https://github.com/microsoft/PowerToys/issues/2106.

This PR adds `is_process_elevated()` caching.